app/component/toast.tsx

```diff
@@ -20,8 +16,21 @@ const statusTextColor: Record<ToastStatus, string> = {
   failed: "text-[#FF9393]",
   sent: "text-[#7EE0FF]"
 }
-export const Toast = (props: ToastProps) => {
-  const { status, message } = props
+export const Toast = () => {
+  const { toast, setToast } = useContext(ToastContext)
+  const { status, message } = toast
+
+  useEffect(() => {
+    if (toast && toast.message && toast.status) {
+      const timer = setTimeout(() => {
+        setToast(null, null)
+      }, 2000)
+      return () => clearTimeout(timer)
+    }
+  }, [toast, setToast])
```

Comment:
Q: I think `setToast` doesn't need to put into `dep`?

Reply:
you're right. The set functions returned by useState is stable, so there's no need to add it into dep array.

[remove setToast from dep](https://github.com/pilagod/waallet-extension/pull/158/commits/8aeadf276edf68f6f8d7061bcef9708fa4e7e29e)